sismic/code/evaluator.py

```diff
@@ -88,9 +88,7 @@ def evaluate_guard(self, transition: Transition, event: Optional[Event]=None) ->
         :param event: instance of *Event* if any
         :return: truth value of *code*
         """
-        if transition.guard:
-            return self._evaluate_code(transition.guard, additional_context={'event': event})
-        return None
+        return self._evaluate_code(getattr(transition, 'guard', None), additional_context={'event': event})
```

Comment:
This is not equivalent if `transition.guard` is `None`. In the first case, `_evaluate_code` won't be called, where in the second case, it will be called with `None`. I don't have time to check right now, but either you should not remove the conditional statement, or you should set default value to the empty string?

Reply:
The Python Evaluator **was** doing this:
```
return self._evaluate_code(getattr(transition, 'guard', None),  ...)
```
where the code evaluates to either a string of code or None. That was then handled in the _evaluate_code/_execute_code block with:
```
if code is None:
    return []
```

I thought the implementation in Python Evaluator was more succinct and therefore adopted it to both the Evaluator and EnhancedEvaluator classes so that implementations would be consistent. Perhaps what really needs to happen is to update the abstract function definitions for _evaluate_code() and _execute_code() along these lines

```
@abc.abstractmethod
 def _execute_code(self, code: Optional[str]=None, *, additional_context: Mapping[str, Any]=None) -> List[Event]:
        foo = bar
```
If I have that bit correct ... **Optional[str]=None** ... I will test it later anyway, just to see.

I think None is better than an empty string, since an empty string is valid code, and None indicates _no_ code at all ... but I don't mind either way :-)  I've got the example Lark DSL wired up with Sismic via this mechanism, it works! I hit this problem too ... but what you had in the Python Evaluator was fine by me.

Reply:
You're right ;) Let's continue with `Optional[str]` in both `_execute_code` and `_evaluate_code`.